pinc/archiving.inc

```diff
@@ -26,15 +26,25 @@ function archive_project($project, $dry_run)
     $mod_time_str = strftime('%Y-%m-%d %H:%M:%S', $project->modifieddate);
     echo "$projectid ($mod_time_str) \"$project->nameofwork\"\n";
 
-    if (!does_project_page_table_exist($projectid)) {
+    if (!$project->pages_table_exists) {
         echo "    Table $projectid does not exist.\n";
     } elseif ($dry_run) {
         echo "    Move table $projectid to $archive_db_name.\n";
     } else {
-        mysqli_query(DPDatabase::get_connection(), "
+        $sql = "
             ALTER TABLE $projectid
```

Comment:
you escape projectid below, but not here. Is it intentional?

Reply:
Because this is a table name, not a string. See [Validating project tables](https://www.pgdp.net/wiki/DP_Code_Best_Practices#Validating_project_tables) in our best practices. In this case the only caller is pulling the fields from the DB so we know they are valid projectids and thus valid tables names.